src/main/kotlin/com/getkeepsafe/dexcount/Tasks.kt

```diff
@@ -165,20 +168,38 @@ abstract class DexMethodCountTaskBase: DefaultTask() {
      * counts of the current dex/apk file.
      */
     private fun generatePackageTree() {
-        val file = fileToCount ?: throw AssertionError("file is null: rawInput=${rawInputRepresentation}")
+        val file = fileToCount ?: throw AssertionError("file is null: rawInput=$rawInputRepresentation")
+
+        isAndroidProject = file.extension == "aar" || file.extension == "apk"
+        val isJar = file.extension == "jar"
+
+        if ((!isAndroidProject && !isJar) || (isAndroidProject && isJar)) {
+            throw IllegalStateException("File extension is unclear: $file")
+        }
 
         startTime = System.currentTimeMillis()
 
-        val dataList = DexFile.extractDexData(file, config.dxTimeoutSec)
+        val dataList = if (isAndroidProject) DexFile.extractDexData(file, config.dxTimeoutSec) else emptyList()
+        val jarFile = when {
+            isAndroidProject && config.printDeclarations -> JarFile.extractJarFromAar(file)
```

Comment:
This won't work for APKs, for which `isAndroidProject` will be true.  I see that `printDeclarations` is disallowed for applications, but IMO a more-robust check here would be `isAar` (for example, in case the policy in the plugin changes).

Reply:
You're right, I make this more explicit.